components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.ts

```diff
@@ -0,0 +1,221 @@
+  public server: Server;
+  public orgs: Org[] = [];
+  public tabValue: ChefServerTabName = 'orgs';
+  public url: string;
+  public updateServerForm: FormGroup;
+  public orgForm: FormGroup;
+  public createModalVisible = false;
+  public creatingServerOrg = false;
+  public conflictErrorEvent = new EventEmitter<boolean>();
+  public orgToDelete: Org;
+  public deleteModalVisible = false;
+  public modalType: string;
+  private id: string;
+  public saveSuccessful = false;
+
+  // isLoading represents the initial load as well as subsequent updates in progress.
+  public isLoading = true;
+  public saving = false;
+  private isDestroyed = new Subject<boolean>();
+
+  constructor(
+    private fb: FormBuilder,
+    private store: Store<NgrxStateAtom>,
+    private router: Router,
+    private layoutFacade: LayoutFacadeService
+
+  ) {
+
+    this.orgForm = fb.group({
+      name: ['', [Validators.required]],
+      admin_user: ['', [Validators.required]],
+      admin_key: ['', [Validators.required]]
+    });
+  }
+
+  ngOnInit() {
+    this.layoutFacade.showInfastructureSidebar();
+    // Populate our tabValue from the fragment.
+    this.store.select(routeURL).pipe(takeUntil(this.isDestroyed))
+    .subscribe((url: string) => {
+      this.url = url;
+      const [, fragment] = url.split('#');
+      this.tabValue = (fragment === 'details') ? 'details' : 'orgs';
+    });
+
+    this.updateServerForm = this.fb.group({
+      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      description: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      fqdn: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      ip_address: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]]
+    });
+
+    this.store.select(routeParams).pipe(
+      pluck('id'),
+      filter(identity),
+      takeUntil(this.isDestroyed))
+      .subscribe((id: string) => {
+        this.id = id;
+        this.store.dispatch(new GetServer({ id }));
+        this.store.dispatch(new GetOrgsForServer({ server_id: id }));
+      });
+
+      combineLatest([
+        this.store.select(getStatus),
+        this.store.select(getAllOrgsForServerStatus)
+      ]).pipe(
+        takeUntil(this.isDestroyed)
+      ).subscribe(([getServerSt, getOrgsSt]) => {
+          this.isLoading =
+            !allLoaded([getServerSt, getOrgsSt]);
+        });
+
+      combineLatest([
+        this.store.select(getStatus),
+        this.store.select(getAllOrgsForServerStatus),
+        this.store.select(serverFromRoute),
+        this.store.select(allOrgs)
+      ]).pipe(
+          filter(([getServerSt, getOrgsSt, _serverState, _allOrgsState]) =>
+            getServerSt === EntityStatus.loadingSuccess &&
+            getOrgsSt === EntityStatus.loadingSuccess),
+          filter(([_getServerSt, _getOrgsSt, serverState, allOrgsState]) =>
+            !isNil(serverState) && !isNil(allOrgsState)),
```

Comment:
Consider using a single filter to make this more readable.
```
filter(([getServerStatus, getOrgsStatus, serverState, allOrgsState]) =>
  getServerStatus === EntityStatus.loadingSuccess &&
  getOrgsStatus === EntityStatus.loadingSuccess &&
  !isNil(serverState) &&
  !isNil(allOrgsState))
```

Reply:
I have added changes for this.